src/sentry/api_gateway/proxy.py

```diff
@@ -0,0 +1,54 @@
+"""
+Utilities related to proxying a request to a region silo
+"""
+
+from urllib.parse import urljoin
+
+import requests as external_request
+from requests import Response as external_response
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.utils import generate_region_url
+from sentry.models.organization import Organization
+
+
+def get_org_region(slug: str) -> str:
+    from sentry.types.region import get_region_for_organization
+
+    org = Organization.objects.get(slug=slug)
+    region_info = get_region_for_organization(org)
+    return region_info.address
+
+
+def build_url(region: str, request: Request) -> str:
+    """Might need some changes to handle other format region address."""
+    region_url = generate_region_url(region)
+    return urljoin(region_url, request.path)
+
+
+def parse_response(response: external_response) -> Response:
+    """
+    Convert the Responses class from requests into the drf Response
+    """
+    return Response(data=response.json(), headers=response.headers, status=response.status_code)
```

Comment:
Can we construct the Response object _without_ intermediately buffering the json response here?  In fact, can we construct the data with the unopened response body stream somehow?  Or does drf require the object materialized?

I guess this does broadly work and it may be the correct solution, but it seems wasteful.  We're forcing serialization and deserialization twice, thus reducing performance accordingly.

Reply:
👍 to this. I'm 99% certain that Python exposes the necessary streaming interfaces. I can help dig through the library API if you need.

Reply:
Decided to leverage requests streaming flag and  Django's streaminghttpresponse.
This should help avoid large responses in memory.